cmd/convert.go

```diff
@@ -0,0 +1,368 @@
+		return nil
+	}
+
+	fmt.Println("Are you REALLY sure (y/n)?")
+	yn, _ = reader.ReadString('\n')
+	if !(strings.ToLower(yn) == "y\n" || strings.ToLower(yn) == "yes\n" || strings.ToLower(yn)[:1] == "y") {
+		fmt.Println("No changes made")
+		return nil
+	}
+
+	fmt.Println("Working...")
+
+	// Set repo path
+	repoPath, err := repo.GetRepoPath(x.Testnet)
+	if err != nil {
+		return err
+	}
+	if x.DataDir != "" {
+		repoPath = x.DataDir
+	}
+
+	// Wipe database tables
+	var dbPath string
+	if x.Testnet {
+		dbPath = path.Join(repoPath, "datastore", "testnet.db")
+	} else {
+		dbPath = path.Join(repoPath, "datastore", "mainnet.db")
+	}
+	sqlitedb, err := sql.Open("sqlite3", dbPath)
+	if err != nil {
+		return err
+	}
+	if x.Password != "" {
+		p := "pragma key='" + x.Password + "';"
+		sqlitedb.Exec(p)
+	}
+
+	_, err = sqlitedb.Exec("DELETE FROM txns;")
+	if err != nil {
+		return err
+	}
+	_, err = sqlitedb.Exec("DELETE FROM stxos;")
+	if err != nil {
+		return err
+	}
+	_, err = sqlitedb.Exec("DELETE FROM utxos;")
+	if err != nil {
+		return err
+	}
+
+	// Update coin type in config file
+	cf, err := ioutil.ReadFile(path.Join(repoPath, "config"))
+	if err != nil {
+		return err
+	}
+	var cfgIface interface{}
+	json.Unmarshal(cf, &cfgIface)
+	cfgObj, ok := cfgIface.(map[string]interface{})
+	if !ok {
+		return errors.New("Invalid config file")
+	}
+
+	walletIface, ok := cfgObj["Wallet"]
+	if !ok {
+		return errors.New("Config file missing wallet field")
+	}
+	walletCfg, ok := walletIface.(map[string]interface{})
+	if !ok {
+		return errors.New("Invalid config file")
+	}
+	walletCfg["Type"] = cfgtype
+	if strings.ToLower(args[0]) == "zcash" {
+		walletCfg["Binary"] = args[1]
+	}
+	out, err := json.MarshalIndent(cfgObj, "", "   ")
+	if err != nil {
+		return err
+	}
+	f, err := os.Create(path.Join(repoPath, "config"))
+	if err != nil {
+		return err
+	}
+	_, err = f.Write(out)
+	if err != nil {
+		return err
+	}
+	f.Close()
+
+	// Update listings with new coin type
+	creationDate := time.Now()
+	var sqliteDB *db.SQLiteDatastore
+	sqliteDB, err = InitializeRepo(repoPath, x.Password, "", x.Testnet, creationDate)
+	if err != nil && err != repo.ErrRepoExists {
+		return err
+	}
+
+	// If the database cannot be decrypted, exit
+	if sqliteDB.Config().IsEncrypted() {
+		sqliteDB.Close()
+		fmt.Print("Database is encrypted, enter your password: ")
+		bytePassword, _ := terminal.ReadPassword(int(syscall.Stdin))
+		fmt.Println("")
+		pw := string(bytePassword)
+		sqliteDB, err = InitializeRepo(repoPath, pw, "", x.Testnet, time.Now())
+		if err != nil && err != repo.ErrRepoExists {
+			return err
+		}
+		if sqliteDB.Config().IsEncrypted() {
+			PrintError("Invalid password")
+			os.Exit(3)
+		}
+	}
+	// Get current identity
+	identityKey, err := sqliteDB.Config().GetIdentityKey()
+	if err != nil {
+		PrintError(err.Error())
+		return err
+	}
+	identity, err := ipfs.IdentityFromKey(identityKey)
+	if err != nil {
+		return err
+	}
+
```

Comment:
`	// Check for old moderators
	settings, err := sqliteDB.Settings().Get()
	if err != nil && err != db.SettingsNotSetError {
		log.Error(err)
		return err
	}

	if len(*settings.StoreModerators) > 0 {
        fmt.Println("Opps!, it looks like you still have moderators using your old currency")
        fmt.Println("You'll have to remove them manually from their home pages, they will not appear in the client.")
        for _, mid := range *settings.StoreModerators {
            fmt.Printf("  ob://%s/home  \n", mid)
		}
	}`

I couldn't see a way to easily modify the settings.  It would be best just to remove them automatically.

Reply:
Gotcha. I think they should still be able to remove them in settings but you're right it probably should clear out of the settings.

Reply:
ATM. In all UI lists, moderators are filtered by accepted currency. If the store is associated with several moderators of a different currency, they do not appear in the moderator list in settings, nor in available moderators in checkout.

The only way to I've found to remove moderators of a different currency is from their home page.

FYI it is also possible to add a moderator of a different currency from their homepage. Which is not a server problem